actin/personalization/fetch_tool_from_registry

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -e
+
+function do_curl() {
+    # New versions (>7.52.1) of `curl` work properly with `--oauth2-bearer` instead of the explicit setting of a header
+    # At time of writing the imager VM was using this older version without prospect of upgrade
+    token="$(gcloud auth print-access-token)"
+    curl -H "authorization: Bearer ${token}" "$@"
+}
+
+AR_URL="https://europe-west4-maven.pkg.dev/actin-build/build-registry-maven/com/hartwig/actin"
+
+[[ $# -ne 3 ]] && echo "USAGE: $0 [tool name] [tool version] [destination path]" && exit 1
+
+tool=$1
+version=$2
+destination=$3
+
+remoteDir="${AR_URL}/${tool}/system/${version}"
+
+expected_md5="$(do_curl -o - -L "${remoteDir}/system-${version}.jar.md5")"
+do_curl -o "${destination}" -L "${remoteDir}/system-${version}.jar"
+actual_md5="$(md5sum "${destination}" | awk '{print $1}')"
```

Comment:
Maybe the md5 check is a little overkill? Makes the script somewhat more complex than it needs to be.

Reply:
As the author of the original script (see https://github.com/hartwigmedical/pipeline5/blob/a6abe77066a247c5d8bc9ec7e2e40b610fbd33fa/image/copy_to_imager_vm/fetch_tool_from_registry.sh) I think I'm in a good position to comment.

Error handling on the Web is anything but consistent and over the years I have found it better to assert against content rather than rely on return codes from `curl`. Also it's not the case that `curl` can necessarily provide useful responses under all circumstances, so the choice was made to be more rather than less careful. This for instance appears in the `curl` manual page:

```
-f, --fail
              (HTTP) Fail silently (no output at all) on server errors. This is mostly done to better enable scripts etc to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an HTML  docu‐
              ment stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.

              This method is not fail-safe and there are occasions where non-successful response codes will slip through, especially when authentication is involved (response codes 401 and 407).
```

That said, the source script was written to be resilient to several failure modes that may not exactly apply here, so it could be reasonable to scale back the paranoia. A simple way to do that which might work for our purposes would be to change `curl` to `curl -f` in `do_curl()` and hull out all this other stuff. But also don't be surprised if it fails in weird ways now or in the future.